### PR TITLE
Add basic circle and helix path creation

### PR DESCRIPTION
**Pull request by k3vnchen**
Title says all. Details below:
- Rename `arcball.py` to `glhelper.py`
- Shape generating (in `glhelper.py`) is very preliminary, and we will want to separate path generating to separate classes

### CODE REVIEW

File: components/camview.py [hunk 1]
```diff
@@ -3,27 +3,28 @@
 
 import numpy as np
 from enums import Axis
+import math
+import time
```
Comment (k3vnchen) [inline components/camview.py:7]: `time` is an unused import, but we may need it to create a render loop

File: components/canvas.py [hunk 2]
```diff
@@ -55,12 +57,11 @@ def processSizeEvent(self, event):
...
-            self.SetCurrent(self.context)
-            self.OnReshape()
-            self.Refresh(False)
-            timer = wx.CallLater(100, self.Refresh)
-            timer.Start()
+        self.SetCurrent(self.context)
+        self.OnReshape()
+        self.Refresh(False)
+        timer = wx.CallLater(100, self.Refresh)
+        timer.Start()
```
Comment (k3vnchen) [inline components/canvas.py:64]: Moving this out of the if statement is what fixed #15

File: components/glhelper.py [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+    phi = 2.0 * math.asin(t)
+
+    return vector_to_quat(rot_axis, phi)
+
+
+def sphere_coords(x, y, r):
+    """Compute the intersection from screen coords to the arcball sphere."""
+    d2 = x*x + y*y
+    r2 = r * r
+    if math.sqrt(d2) <= r * 0.70710678118654752440:
```
Comment (parksong22) [inline components/glhelper.py:36]: What does 0.70710678118654752440 and r * 0.70710678118654752440 represents?

File: components/glhelper.py [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+        x = (x + tx) * radial_factor
+        y = (y + ty) * radial_factor
+        z = (z + tz) * radial_factor
+    return vertices, count
+
+
+def draw_circle_trig(p, n, r, sides=36):
+    """Draw circle given point, normal vector, radius, and # sides."""
+    a, b, n = rotate_basis(*n)
+    tau = 6.28318530717958647692
```
Comment (parksong22) [inline components/glhelper.py:212]: I see this in multiple functions. would it be better to make it a global variable?

File: components/camview.py (hunk 1 above)
Comment (k3vnchen) [reply components/camview.py:7]: Gotcha, will remove for now

File: components/glhelper.py (hunk 3 above)
Comment (k3vnchen) [reply components/glhelper.py:36]: This achieves what is described [here](https://www.khronos.org/opengl/wiki/Object_Mouse_Trackball#Of_mice_and_manipulation). To smooth the transition from dragging in the sphere vs out, we define a piecewise function between the sphere and the hyperbolic sheet, where sqrt(x^2+y^2) <= sqrt(r^2/2), and sqrt(r^2/2) = r * 0.707106...

File: components/canvas.py [hunk 5]
```diff
@@ -24,15 +25,16 @@ def __init__(self, parent, *args, **kwargs):
         self.gl_init = False
         self.gl_broken = False
         self.context = glcanvas.GLContext(self)
-        self.context_attrs = glcanvas.GLContextAttrs()
-        self.display_attrs = glcanvas.GLAttributes()
+        context_attrs = glcanvas.GLContextAttrs()
+        context_attrs.CoreProfile().OGLVersion(4, 5).Robust().ResetIsolation().EndList()
+        display_attrs = glcanvas.GLAttributes()
+        display_attrs.PlatformDefaults().MinRGBA(8, 8, 8, 8).DoubleBuffer().Depth(32).EndList()
```
[the remark this answers is not in the document]
Comment (k3vnchen) [reply components/canvas.py:31]: On it

File: components/glhelper.py (hunk 4 above)
Comment (k3vnchen) [reply components/glhelper.py:212]: I'd rather not deal with global variables, so I think it's fine this way.

**Comment by k3vnchen** [comment]
Updated code as per feedback.

**Comment by k3vnchen** [comment]
@parksong22 Finished requested changes

**Pull request merged**